src/Fabulous/WidgetDiff.fs

```diff
@@ -23,6 +23,42 @@ module EnumerationMode =
         | ValueNone, ValueSome next -> EnumerationMode.AllAddedOrRemoved(next, true)
         | ValueSome prev, ValueSome next -> EnumerationMode.ActualDiff(prev, next)
 
+module private SkipRepeatingScalars =
+    /// Context:
+    /// In WidgetBuilders we can easily have duplicate attributes
+    /// like Label("aha!").color("red").color("blue")
+    /// Diffing algorithm already uses stable sort for diffing attributes
+    /// Thus we need to leverage that fact
+    ///
+    /// In particular we might have a bug in this context
+    /// prev:  Label("aha!").color("red").color("blue")
+    /// next:  Label("aha!").color("red")
+    ///
+    /// After the diffing the resulting color should be "red", but previously
+    /// it was interpreted as "removed color property" instead
+    ///
+    /// This functions skips "color("red")" from "color("red").color("blue")"
+    /// Effectively taking only the latest attribute for a particular property
+    ///
+    /// TODO: is there a more optimal way to write it? This is a part of the hot path
+    /// I couldn't detect perf differences with and without this function
+    /// but still might be interesting to tinker about it more
+    let inline skip (scalars: ScalarAttribute array) (pos: int) =
+        let length = scalars.Length
+        // either the last element or out of bounds
+        if pos >= length - 1 then
+            pos
+        else
+            // that means that there is at least one more element ahead
+            let key = scalars.[pos].Key
+            let mutable resultingIndex = pos
+
+            while (length - 1 > resultingIndex)
```

Comment:
Not sure if it makes any difference, but maybe a simple `if scalars.[pos + 1].Key = key then` would be better than a while condition on the hot path?
99.9% of attributes won't have a duplicate attribute.

This is more IL-level optimisation though 😅